hip_data_tools/aws/athena.py

```diff
@@ -78,27 +79,178 @@ def get_table_settings_for_dataframe(dataframe: DataFrame, partitions: dict, s3_
+        partition_fields (string): partition field names
+
+    Returns (string): CTAS Query in a string
+
+    """
+    return _get_ctas_statement(destination_bucket, destination_key, destination_table,
+                               partition_fields, select_query, file_format='TEXTFILE')
+
+
+def generate_parquet_ctas(select_query, destination_table, destination_bucket, destination_key,
+                          partition_fields=''):
+    """
+    Method to generate a CTAS query string for creating parquet output
+
+    Args:
+        select_query (string): the query to be used for table generation
+        destination_table (string): name of the new table being created
+        destination_bucket (string): the s3 bucket where the data from select query will be stored
+        destination_key (string): the s3 directory where the data from select query will be stored
+        partition_fields (string): partition field names
+
+    Returns (string): CTAS Query in a string
+
+    """
+    return _get_ctas_statement(destination_bucket, destination_key, destination_table,
+                               partition_fields, select_query, file_format='parquet')
+
+
+def _get_ctas_statement(destination_bucket, destination_key, destination_table, partition_fields,
+                        select_query, file_format):
+    delimiter = ''
+    if file_format == 'TEXTFILE':
+        delimiter = "field_delimiter=',',"
+    partitioned_by = ""
+    if partition_fields != '':
+        partitioned_by = """,
+        partitioned_by = ARRAY[{partition_keys}]
+        """.format(partition_keys=partition_fields)
+    final_query = """
+        CREATE TABLE {destination_table}
+        WITH (
+            {delimiter}
+            format='{file_format}',
+            external_location='s3://{bucket}/{key}'{partitioned_by}
+        ) AS
+        {athena_query}
+        """.format(destination_table=destination_table,
+                   bucket=destination_bucket,
+                   key=destination_key,
+                   athena_query=select_query,
+                   partitioned_by=partitioned_by,
+                   file_format=file_format,
+                   delimiter=delimiter)
+    return final_query
+
+
+def zip_columns(column_list):
+    """
+    Combine the column list into a zipped comma separated list of column name and data type
+    Args:
+        column_list (list): an array of dictionaries with keys column and type
+
+    Returns (string): a string containing comma separated list of column name and data type
+
+    """
+    return ", ".join(["{} {}".format(col['column'], col["type"]) for col in column_list])
+
+
+def _construct_table_partition_ddl(partitions):
+    partition_query = ""
+    if partitions:
+        partition_query = """
+        PARTITIONED BY ( 
+          {columns}
+          )
+          """.format(columns=zip_columns(partitions))
+    return partition_query
+
+
+def _construct_table_exists_ddl(enable_exists):
```

Comment:
This can be a todo for future, or if easy enough, refactor now. will also help in unit tests and stubbing:
AthenaQueryConstructor
    -> AthenaCTASQueryConstructor
    -> AthenaCreateTableQueryConstructor
    -> AthenaAddPartitionQueryConstructor
    -> AthenaRemovePartitionQueryConstructor

Reply:
I'd rather do this refactoring in a separate PR. I think this PR is already massive enough.

Reply:
Changed the function name